care/emr/api/viewsets/totp.py

```diff
@@ -0,0 +1,334 @@
+        responses={
+            200: TOTPSetupResponse,
+            400: {"type": "object", "properties": {"error": {"type": "string"}}},
+        },
+    )
+    @action(detail=False, methods=["POST"])
+    def setup(self, request):
+        user = request.user
+
+        mfa_settings = user.mfa_settings or {}
+
+        if self._check_totp_enabled(mfa_settings) is None:
+            return Response(
+                {
+                    "error": "Two-factor authentication is already enabled for your account"
+                },
+                status=status.HTTP_400_BAD_REQUEST,
+            )
+
+        secret = random_base32()
+        encrypted_secret = encrypt_string(secret)
+
+        totp = TOTP(secret)
+        uri = totp.provisioning_uri(name=user.email, issuer_name="CARE")
+
+        user.totp_secret = encrypted_secret
+        user.save(update_fields=["totp_secret"])
+
+        response_data = TOTPSetupResponse(uri=uri, secret_key=secret)
+        return Response(response_data.model_dump())
+
+    @staticmethod
+    def _generate_backup_codes(count: int = 10) -> list[str]:
+        """Generate 8-digit backup codes."""
+        codes = []
+        for _ in range(count):
+            code = "".join(choice(digits) for _ in range(8))
+            codes.append(code)
+        return codes
+
+    @extend_schema(
+        description="Verify TOTP code and enable 2FA",
+        request=TOTPVerifyRequest,
+        responses={
+            200: TOTPVerifyResponse,
+            400: {"type": "object", "properties": {"error": {"type": "string"}}},
+        },
+    )
+    @action(detail=False, methods=["POST"])
+    def verify(self, request):
+        request_data = TOTPVerifyRequest(**request.data)
+        user = request.user
+
+        if not user.totp_secret:
+            return Response(
+                {"error": "TOTP not configured"}, status=status.HTTP_400_BAD_REQUEST
+            )
+
+        secret = decrypt_string(user.totp_secret)
+        totp = TOTP(secret)
+
+        if totp.verify(request_data.code):
+            backup_codes = self._generate_backup_codes()
+
+            mfa_settings = user.mfa_settings or {}
+            mfa_settings["totp"] = {
+                "enabled": True,
+                "enabled_at": timezone.now().isoformat(),
+                "backup_codes": [
+                    {
+                        "code": make_password(code),
+                        "used": False,
+                        "created_at": timezone.now().isoformat(),
+                    }
+                    for code in backup_codes
+                ],
+            }
+            user.mfa_settings = mfa_settings
+            user.save(update_fields=["mfa_settings"])
+
+            send_totp_enabled_email.delay(user.email, user.username)
+
+            response_data = TOTPVerifyResponse(
+                message="Two-factor authentication has been enabled successfully. Please save your backup codes in a secure location.",
```

Comment:
I dont think the backend should send any message, the message should be assumed based on the response, let the FE maintain the message.